kdc/kerberos5.c

```diff
@@ -2146,6 +2147,8 @@ _kdc_as_rep(astgs_request_t r)
 		continue;
 	    if (r->armor_crypto == NULL && (pat[n].flags & PA_REQ_FAST))
 		continue;
+	    if (r->armor_crypto != NULL && !(pat[n].flags & PA_REQ_FAST))
+		continue;
```

Comment:
Won't this change break mechanisms like GSS that can work both with and without FAST?

Reply:
Did you mean to check for `PA_NOT_IN_FAST` instead?

Reply:
Right, I neglected the GSS padata type. Is it better with the changes I've now pushed?

Reply:
Is there a reason we don't want to allow PKINIT inside FAST? cc @nicowilliams

Reply:
I don't see any reason to disallow PKINIT in FAST.